packages/fields/src/fields/featured-image/style.scss

```diff
@@ -83,13 +83,10 @@ fieldset.fields-controls__featured-image {
 }
 
 .dataviews-view-table__cell-content-wrapper {
-	.fields-controls__featured-image-image {
-		width: 32px;
-		height: 32px;
-	}
-
+	.fields-controls__featured-image-image,
 	.fields-controls__featured-image-placeholder {
 		width: 32px;
 		height: 32px;
+		display: block;
```

Comment:
Similar to the previous bug. The lack of `display` property was creating whitespace beneath the featured image which messed up alignment.